manyfews/calculations/zentra.py

```diff
@@ -122,6 +123,114 @@ def zentraReader(startTime, endTime, stationSN):
+
+    for data in zentraReadingData:
+        RHList.append(data.relative_humidity)
+        precipList.append(data.precipitation)
+        airTemList.append(data.air_temperature)
+        wSpeedList.append(data.wind_speed)
+        wDirectionList.append(data.wind_direction)
+
+    # convert temperature Unit from °C to ℉
+    airTemList = [i + 273.15 for i in airTemList]
+
+    ##############################################
+    #  filter the wrong data (Temporary solution)
+    #############################################
+    wSpeedList = [
+        1 if i == None else i for i in wSpeedList
+    ]  # for None data, set it to default (1)
+    wDirectionList = [
+        1 if i == None else i for i in wDirectionList
+    ]  # for None data, set it to default (1)
+    airTemList = [
+        1 if i == None else i for i in airTemList
+    ]  # for None data, set it to default (1)
+    precipList = [
+        1 if i == None else i for i in precipList
+    ]  # for None data, set it to default (1)
+    RHList = [
+        1 if i == None else i for i in RHList
+    ]  # for None data, set it to default (1)
+    ###############################################
+
+    zentraReadingList = list(
+        zip(RHList, precipList, airTemList, wSpeedList, wDirectionList)
+    )
+
+    zentraData = np.array(zentraReadingList)
+
+    # Convert wind speed and wind direction into Uwind and Vwind
+    wSpeed = zentraData[:, 3]
+    wDirection = zentraData[:, 4]
+
+    # calculation of Uwind and Vwind
+    wDirection = (wDirection * (math.pi)) / 180  # Convert angle to magnitude
+
+    uWind = (np.cos(wDirection)) * wSpeed
+    vWind = (np.sin(wDirection)) * wSpeed
+    zentraData[:, 3] = uWind
+    zentraData[:, 4] = vWind
+
+    # aggregate zentra data within every 6(dt) hours: 6 hours (dt =0.25 unit:day)
+    dt = float(settings.MODEL_TIMESTEP)
+    intervalNum = int(1 / dt)  # the number of aggregating points per day.
+    dataPointNum = int(
+        (len(zentraData)) / intervalNum
+    )  # the number of data sets per aggregation
+
+    aggregatedData = np.zeros([intervalNum, 6], dtype=float)
+    for i in range(intervalNum):
+
+        # get the array sequence of data for each aggregating period
+        seq = range(i * dataPointNum, ((i + 1) * dataPointNum))
+        intervalData = zentraData[np.array(tuple(seq))]
+        aggregatedData[i, 0] = np.mean(intervalData[:, 0])  # mean result of RH
+        aggregatedData[i, 3] = np.mean(intervalData[:, 3])  # mean result of uWind
+        aggregatedData[i, 4] = np.mean(intervalData[:, 4])  # mean result of vWind
+        aggregatedData[i, 5] = np.sum(intervalData[:, 1])  # sum result of precipitation
+        aggregatedData[i, 1] = np.max(
+            intervalData[:, 2]
+        )  # find the maximum temperature
+        aggregatedData[i, 2] = np.min(
+            intervalData[:, 2]
+        )  # find the minimum temperature
+
+    from django.contrib.gis.geos import Point
+
+    location = Point(0, 0)
```

Comment:
Can you update this to get the location from `ZentraDevice` of the readings? You should be able to do something like:

```
location = zentraReadingData[0].device.location
```

Reply:
corrected.